src/frontend/src/common/analytics/eventTypes.ts

```diff
@@ -4,11 +4,168 @@
+ * - All the keys should be in **snake_case**.
+ * - The structure should always be a **flat object**. (extends `EventData`)
+ * - If you must send more complicated info (say, an array of IDs), convert
+ *   it into a string (JSON.stringify, probably) and send the string.
+ * - Always use `type`, never `interface`. (Necessary for generic extends)
+ *   See https://github.com/microsoft/TypeScript/issues/15300#issuecomment-371353444
+ * --- RECOMMENDATIONS for adding new event TS types ---
+ * - Avoid optional keys or using value of `undefined` unless necessary.
+ * - Document what each key means in the context of that event type.
+ * - If a key holds JSON, use `JsonString` type and document what it contains.
+ * - Try to avoid sending JSON strings if you can easily break it into multiple
+ *   keys with simple values instead.
+ * - When developing, try to avoid actually sending payload to Segment until
+ *   you are confident in the structure for that event type: if you send it
+ *   prematurely, you might have to wipe out that event's corresponding table
+ *   due to schema weirdness which can be a hassle to deal with when testing.
+ */
+
+/** EVENT_TYPES.TREE_VIEW_NEXTSTRAIN */
+export type AnalyticsTreeViewNextstrain = {
+  // Tree that user is being sent to view
+  tree_id: number;
+};
+
+/** EVENT_TYPES.TREE_CREATION_NEXTSTRAIN */
+export type AnalyticsTreeCreationNextstrain = {
+  // PK of the workflow that kicks off the creation of requested tree
+  // Note: creating a tree kicks off process, it could still fail in pipeline.
+  phylo_run_workflow_id: number;
+  // Type of tree being created
+  tree_type: string;
+};
+
+/** EVENT_TYPES.TREE_DOWNLOAD_TREE_FILE */
+export type AnalyticsTreeDownloadTreeFile = {
+  // Tree user is downloading.
+  // null indicates tree does not exist, but should be impossible if user
+  // is downloading it. Mostly here to make TS happy, but if we ever get null
+  // in this event for tree_id, it very likely indicates a bug with app.
+  tree_id: number | null;
+  // PK of the workflow that kicked off the creation of this tree.
+  // Should never be null, but TS for underlying item does not guarantee it, so
+  // the null possibility is mostly to keep TS happy. If null, app has a bug.
+  phylo_run_workflow_id: number | null;
+  // User can download tree with samples using either their Private IDs or
+  // their Public IDs.
+  sample_id_type: "PRIVATE" | "PUBLIC";
+};
+
+/** EVENT_TYPES.TREE_DOWNLOAD_SELECTED_SAMPLES_TEMPLATE */
+export type AnalyticsTreeDownloadSelectedSamplesTemplate = {
+  // Tree user is downloading template in regards to.
+  // Can download template before tree done or tree failed. Null indicates such
+  tree_id: number | null;
+  // PK of the workflow that kicked off the creation of this tree.
+  // Should never be null, but TS for underlying item does not guarantee it, so
+  // the null possibility is mostly to keep TS happy. If null, app has a bug.
```

Comment:
I'm not sure all the extra `| null` is the best solution for "not technically guaranteed" types here. Is there a reason we can't, for example, check for null-ish values and only track if it exists? or add some logic guards to make TS happy? 

If something here could indicate the app potential has a type-related bug, I would be interested in having the code changed to prevent that possibility from a TS perspective. The point of TS is to catch and prevent these types of mistakes, so adding `| null` seems counter to having TS in the app at all.

Reply:
In general, I agree about the type-related bug, but I didn't really want to go back and figure out why `id?:` is in PhyloRun right now. I think it doesn't make sense for a PhyloRun to be missing an id, but being sure about that was outside what I wanted to figure out right now.

For the TSV download event, I actually disagree: it's intentional for the `tree_id` to be null in the case when the tree is not completed yet or failed, but the user can still download the associated TSV. In that case, the tree_id really should be null. The other event of downloading a tree really should never have a null `tree_id`, so I agree on that count, but fixing it felt like it would take a decent amount of effort since the underlying PhyloRun has no requirement that a tree exists yet. It really comes down to `disabled` being used as form of type guard that I can't easily expand to the analytics call. (Although that's not even a "real" type guard, because we're assuming a contract of how the BE returns `TREE_STATUS.Completed` in terms of the tree, but that feels reasonable to me.)